src/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -50,42 +52,44 @@ const DiffExprCompute = (props) => {
     const numMetadata = Object.values(properties).filter((o) => o.type === 'metadataCategorical').length;
 
     if (!numMetadata) {
-      setType(ComparisonType.within);
+      dispatch(setComparisonType(ComparisonType.within));
     }
     setHasMetadata(numMetadata > 0);
 
-    // Update the hierarchies if a previously selected set were to be deleted.
-    setSelectedGroups(_.mapValues(selectedGroups, (cellSetKey) => {
-      if (!cellSetKey) {
-        return null;
-      }
-
-      const splitKey = cellSetKey.split('/')[1];
+    // If any key in group option is deleted, set option to null
+    Object.keys(comparisonGroup).forEach((type) => {
+      const groupDiff = {};
 
-      if (cellSetKey === 'all' || cellSetKey === 'background' || splitKey === 'rest') {
-        return cellSetKey;
-      }
-
-      if (!properties[splitKey]) {
-        return null;
```

Comment:
why is this code no longer needed?

Reply:
This was previous implementation for checking if a selected key was deleted. This is deleted because the way the selected keys are stored in has changed, from using state to redux and also a deeper object with each state (between/within) storing its own keys. Previously, there was only one.